src/assets/js/ledyer-payments.js

```diff
@@ -0,0 +1,274 @@
+
+            if ( authResponse ) {
+                // if status is authorized, the order is ready to be created
+                if ( authResponse.state === "authorized" ) {
+                    // Get the authorization token to create an order from your backend
+                    const authToken = authResponse.authorizationToken
+                    const { state } = authResponse
+                    const { createOrderUrl, createOrderNonce } = gatewayParams
+
+                    $.ajax( {
+                        type: "POST",
+                        url: createOrderUrl,
+                        dataType: "json",
+                        data: {
+                            state,
+                            order_key: orderId,
+                            auth_token: authToken,
+                            nonce: createOrderNonce,
+                        },
+                        success: ( data ) => {
+                            const {
+                                data: { location },
+                            } = data
+                            window.location = location
+                        },
+                        error: ( jqXHR, textStatus, errorThrown ) => {
+                            console.debug( "Error:", textStatus, errorThrown )
+                            console.debug( "Response:", jqXHR.responseText )
+
+                            submitOrderFail(
+                                "createOrder",
+                                "The payment was successful, but the order could not be created.",
+                            )
+                        },
+                    } )
+                } else if ( authResponse.state === "awaitingSignatory" ) {
+                    const { pendingPaymentUrl, pendingPaymentNonce } = gatewayParams
+                    $.ajax( {
+                        type: "POST",
+                        url: pendingPaymentUrl,
+                        dataType: "json",
+                        data: {
+                            order_key: orderId,
+                            nonce: pendingPaymentNonce,
+                        },
+                        success: ( data ) => {
+                            const {
+                                data: { location },
+                            } = data
+                            window.location = location
+                        },
+                        error: ( jqXHR, textStatus, errorThrown ) => {
+                            console.debug( "Error:", textStatus, errorThrown )
+                            console.debug( "Response:", jqXHR.responseText )
+
+                            submitOrderFail(
+                                "pendingPayment",
+                                "The payment is pending payment. Failed to redirect to order received page.",
+                            )
+                        },
+                    } )
+                }
```

Comment:
I would recommend to split these different cases into their own functions to avoid having a big function here.